src/nbl/video/ILogicalDevice.cpp

```diff
@@ -444,6 +444,21 @@ bool ILogicalDevice::updateDescriptorSets(const std::span<const IGPUDescriptorSe
     return true;
 }
 
+bool ILogicalDevice::nullifyDescriptors(const std::span<const IGPUDescriptorSet::SDropDescriptorSet> dropDescriptors, asset::IDescriptor::E_TYPE descriptorType)
+{
+    for (const auto& drop : dropDescriptors)
+    {
+        auto ds = drop.dstSet;
+        if (!ds || !ds->wasCreatedBy(this))
+            return false;
+
+        ds->dropDescriptors(drop, descriptorType);
```

Comment:
do all validation first, then drop them, right now if you have a validation error midway you'll end up dropping some and not others